TreeTracker/UI/Views/Home/HomeViewModel.swift

```diff
@@ -58,6 +62,17 @@ class HomeViewModel {
     func addTreeSelected() {
         coordinatorDelegate?.homeViewModel(self, didSelectAddTreeForPlanter: planter)
     }
+
+    func fetchProfileData() {
+        selfieService.fetchSelfie(forPlanter: planter) { (result) in
+            switch result {
+            case .success(let data):
+                viewDelegate?.homeViewModel(self, didFetchProfile: data)
+            case .failure:
+                viewDelegate?.homeViewModel(self, didFetchProfile: Asset.Assets.person.image.jpegData(compressionQuality: 1.0)!)
```

Comment:
I think SwiftLint will throw a warning here due to the force unwrap, I agree with the rule around this as we wouldn't want to cause a crash.

I wonder if the `didFetchProfile` delegate function would be better returning a `UIImage` rather than `Data`. There is the argument that UIKit elements shouldn't be in a view model but I don't really agree with it. That way the view isn't responsible for the jpeg decoding either.

Interested in your thoughts here.

Reply:
hmm i really like the idea that the `ViewModel` only returns data and the controller is in charge of any `UI` components. i think the separation of concerns is worth it imo